model/error/event.go

```diff
@@ -377,20 +365,21 @@ func (k *groupingKey) String() string {
 // same grouping key can be collapsed together.
 func (e *Event) calcGroupingKey() string {
 	k := newGroupingKey()
+	var stacktrace m.Stacktrace
 
-	var st m.Stacktrace
-	if e.Exception != nil {
-		k.add(e.Exception.Type)
-		st = e.Exception.Stacktrace
+	for _, ex := range flattenExceptionChain(e.Exception) {
+		k.add(ex.Type)
```

Comment:
Why is https://github.com/elastic/apm-server/pull/2532/files#diff-52137911864c236a97b14516397a04c6R391 not added for all exception entries but only for the first one, while adding `ex.Type` is done in a loop for all entries?

Reply:
Again for simplicity. I don't see the use case in which we receive chained exception causes only with message and not type. I can also change it if you prefer

Reply:
I would change that or at least add a comment, as the decision seems pretty arbitrary and implicit to me.